src/sidecar/connectionStatusUtils.test.ts

```diff
@@ -31,6 +32,60 @@ describe("connectionEventHandler", () => {
     sandbox.restore();
   });
 
+  for (const action of [
+    ConnectionEventAction.CREATED,
+    ConnectionEventAction.UPDATED,
+    ConnectionEventAction.CONNECTED,
+    ConnectionEventAction.DISCONNECTED,
+  ]) {
+    it(`CCloud connection update should cascade through to call ccloudStateHandling.reactToCCloudAuthState on ${action}`, () => {
+      // connectionEventHandler() should call through to ccloudStateHandling.reactToCCloudAuthState()
+      // upon reciept of a connection event for the CCloud connection.
+
+      // Arrange
+      const reactToCCloudAuthStateStub = sandbox.stub(
+        ccloudStateHandling,
+        "reactToCCloudAuthState",
+      );
+
+      const testConnectionEvent: ConnectionEventBody = {
+        action: action,
+        connection: TEST_CCLOUD_CONNECTION,
+      };
+
+      // Act
+      connectionEventHandler(testConnectionEvent);
+
+      // Assert
+      assert.strictEqual(
+        reactToCCloudAuthStateStub.calledOnce,
+        true,
+        "reactToCCloudAuthState called",
+      );
+
+      assert.ok(
+        reactToCCloudAuthStateStub.calledWith(TEST_CCLOUD_CONNECTION),
+        "called with test ccloud connection",
+      );
+    });
+  }
+
+  it("CCloud connection DELETED event should not call ccloudStateHandling.reactToCCloudAuthState", () => {
```

Comment:
new test proving that when we get a DELETED ccloud connection event, we don't react at websocket level. Other side effects (secret store eventing) will kick off the teardown of our ccloud connection.

The issue is that a DELETED event carries the connection spelled as it was right prior to the deletion, so passing it through as received from the websocket layer is wrong, because it doesn't describe reality anymore. Alas that that websocket message didn't just carry the id of the now-gone connection as opposed to an entire Connection structure.